### PR TITLE
Magento 2 1 1 scripted installation

### PR DESCRIPTION
**Pull request by ttnnkkrr**

### CODE REVIEW

File: HPS_Installer.php [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+    echo [\$(date --rfc-3339=seconds)] - Found \$Magento2 >> \${curdir}/HPS_Heartland.log
+
+    Magento2Base=${magentoBaseDir}
+
+    echo "Magento Base Directory Found: \$Magento2Base"
+    echo "[\$(date --rfc-3339=seconds)] - Magento Base Directory Found: \$Magento2Base" >> \${curdir}/HPS_Heartland.log
+
+    echo "Downloading HPS_Heartland from github"
+    echo "[\$(date --rfc-3339=seconds)] - Downloading HPS_Heartland from github" >> \${curdir}/HPS_Heartland.log
+    git clone -b Magento-2-1-1-scripted-installation https://github.com/hps/heartland-magento2-module.git  2>> \${curdir}/HPS_Heartland.log
```
Comment (ttnnkkrr) [inline HPS_Installer.php]: need to take Branch name out of this before merge. Here for testing

File: HPS/Heartland/Model/Payment.php [hunk 2]
```diff
@@ -239,12 +240,12 @@ public function capture(\Magento\Payment\Model\InfoInterface $payment, $amount)
 
             $this->log($response,'setStatus ');
             /** @var \Magento\Sales\Model\Order\Payment $payment */
-            $payment->setStatus($response->responseText)
-                ->setTransactionId($response->transactionId)
-                ->setIsTransactionClosed(false)
-                ->setCcLast4($CcL4)
-                ->setAdditionalInformation($response->authorizationCode)
-                ->setAmount($amount);
+            @$payment->setStatus($response->responseText);
```
Comment (ttnnkkrr) [inline HPS/Heartland/Model/Payment.php:243]: Probably works fine the other way but i noticed it behaved unexpectedly with invalid keys

File: HPS_Installer.php [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+// information about the process user
+$cu                             = UserInfo::currentProcessUser();
+$cug                            = $cu->getUserName() . ":" . $cu->getGroupName();
+
+// information about the Linux Distro
+$LinuxOSInfo                    = new LinuxOSInfo;
+$meminfo                        = (int) floor(preg_replace('/[\D]/','',exec('cat /proc/meminfo | grep MemTotal'))) ; //2097152 about 2 GB
+
+// information about the PHP settings
+$minVersion                     = (PHP_MAJOR_VERSION . PHP_MINOR_VERSION > 55) && (PHP_RELEASE_VERSION > 1);
```
Comment (ttnnkkrr) [inline HPS_Installer.php]: Since the script otherwise works on 2.0.7 I will reduce this restriction to 5.5.22

**Review by slogsdon — APPROVED** [review]
looks good to me

**Pull request merged**